lib/Source.php

```diff
@@ -66,6 +66,41 @@ public function save($opts = null)
 
     /**
      * @param array|null $params
+     * @param array|string|null $opts
+     *
+     * @return Source The deleted source.
+     */
+    public function delete($params = null, $options = null)
+    {
+        self::_validateParams($params);
+
+        $id = $this['id'];
+        if (!$id) {
+            $class = get_class($this);
+            $msg = "Could not determine which URL to request: $class instance "
+             . "has invalid ID: $id";
+            throw new Error\InvalidRequest($msg, null);
+        }
```

Comment:
Thanks!

> If you want, I can refactor to avoid duplicating the code, or just get rid of the check entirely as it doesn't seem terriblye useful.

I'd say that it's okay given that it was probably added at some point for some weird error condition. Let's just merge as is.